src/containers/my-cooperations/empty-cooperation-activities/EmptyCooperationActivities.tsx

```diff
@@ -1,118 +1,68 @@
-    openMenu(event)
-    setIsMenuOpen((prevState) => !prevState)
-  }
-
-  const openAddCourseTemplateModal = () => {
-    closeMenu()
-    openModal({
-      component: <AddCourseTemplateModal closeModal={closeModal} />
-    })
-  }
-
-  const menuIcon = isMenuOpen ? (
-    <KeyboardArrowUpIcon />
-  ) : (
-    <KeyboardArrowDownIcon />
-  )
-
-  const handleFromScratch = () => {
-    closeMenu()
-    setIsActivityCreated(true)
-  }
-
-  const menuItems = [
-    {
-      id: 1,
-      label: <>{t('cooperationsPage.manyTypes.courseTemplate')}</>,
-      icon: <ViewComfyOutlinedIcon />,
-      onClick: openAddCourseTemplateModal
-    },
-    {
-      id: 2,
-      label: <>{t('cooperationsPage.manyTypes.scratch')}</>,
-      icon: <Crop75Icon />,
-      onClick: handleFromScratch
-    }
-  ]
 
-  const menu = menuItems.map(({ id, label, icon, onClick }) => (
-    <MenuItem key={id} onClick={onClick} sx={styles.menuItem}>
-      {icon} {label}
-    </MenuItem>
-  ))
-
-  const createDescriptionSpan = (text: string) => (
+  const createDescriptionSpan = ({
+    text,
+    context,
+    from = 'description'
+  }: {
+    text: string
+    context?: string
+    from?: string
+  }) => (
     <Typography component={ComponentEnum.Span} sx={styles.weightBox}>
-      {t(`cooperationsPage.description.${text}`)}
+      {t(`cooperationsPage.${from}.${text}`, { context })}
     </Typography>
   )
 
-  const componentDescription = (
-    <>
-      {t('cooperationsPage.description.existingCourse')}
-      {createDescriptionSpan('courseTemplate')}
-      {t('cooperationsPage.description.resourceLibrary')}
-      {createDescriptionSpan('module')}
-      {t('cooperationsPage.description.fillThis')}
-      {createDescriptionSpan('lessons')}
-      {t('cooperationsPage.description.or')}
-      {createDescriptionSpan('quizzes')}
-      {t('cooperationsPage.description.resourcesLibrary')}
-    </>
-  )
-
-  const startIcon = <AddIcon fontSize={SizeEnum.Small} />
-
-  const menuCondition = renderMenu(menu)
-
+  const componentDescription: Partial<
+    Record<UserRoleEnum, string | ReactElement>
+  > = {
+    [UserRoleEnum.Tutor]: (
+      <>
+        {t('cooperationsPage.description.existingCourse')}
+        {createDescriptionSpan({ text: 'courseTemplate' })}
+        {t('cooperationsPage.description.resourceLibrary')}
+        {createDescriptionSpan({ text: 'module' })}
+        {t('cooperationsPage.description.fillThis')}
+        {createDescriptionSpan({ text: 'lessons' })}
+        {t('cooperationsPage.description.or')}
+        {createDescriptionSpan({ text: 'quizzes' })}
+        {t('cooperationsPage.description.resourcesLibrary')}
+      </>
+    ),
+    [UserRoleEnum.Student]: (
+      <>
+        {t('cooperationsPage.description.studentEmptyCooperation')}{' '}
+        {createDescriptionSpan({
+          text: 'notes',
+          from: 'details',
+          context: 'where'
+        })}
```

Comment:
Seems like a lot a repeated code. I suggest creating objects like following:

```jsx
const studentComponentDescriptions = [
    {
         translationKey: "",
         descriptionOptions: {...}
    },
    ....
];

const tutorComponentDescriptions = [
    {
         translationKey: "",
         descriptionOptions: {...}
    },
    ....
]
```

Then you can map over these arrays and reuse this code

I'm not sure about name of the arrays, It requires deeper diving into context of this task

Reply:
Refactored this spot, based on your observasion!

Reply:
Now it look much cleaner and readable. Great job!